### PR TITLE
movingSum/Avg window functions for numeric and decimals

### PR DESCRIPTION
**Pull request by inv2004**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):

Aggregate functions `groupArrayMovingSum(win_size)(x)` and `groupArrayMovingAvg(win_size)(x)`, which calculate moving sum/avg with or without window-size limitation.

Detailed description (optional):

win_size - parameter for limit window size, if not set - it sums all values.

Result: array of the same length and type as x rows.

Example:
```sql
CREATE TABLE t ENGINE = Memory AS SELECT number % 2 as k, 1+number v FROM system.numbers limit 10;

SELECT k, groupArrayMovingSum(3)(v) FROM t GROUP BY k;
┌─k─┬─groupArrayMovingSum(3)(v)─┐
│ 0 │ [1,4,9,15,21]   │
│ 1 │ [2,6,12,18,24]  │
└───┴─────────────────┘

SELECT k, groupArrayMovingSum(v) FROM t GROUP BY k;
┌─k─┬─groupArrayMovingSum(v)───┐
│ 0 │ [1,4,9,16,25]  │
│ 1 │ [2,6,12,20,30] │
└───┴────────────────┘
```

### CODE REVIEW

File: dbms/src/AggregateFunctions/AggregateFunctionMovingSum.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+{
+    assertUnary(name, argument_types);
+
+    bool limit_size = false;
+
+    UInt64 max_elems = std::numeric_limits<UInt64>::max();
+
+    if (parameters.empty())
+    {
+	    // cumulative sum without parameter
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionMovingSum.cpp]: Indent.

File: dbms/src/AggregateFunctions/AggregateFunctionMovingSum.h [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+#include <Columns/ColumnVector.h>
+#include <Columns/ColumnArray.h>
+
+#include <Common/ArenaAllocator.h>
+
+#include <AggregateFunctions/IAggregateFunction.h>
+
+#include <type_traits>
+
+#define AGGREGATE_FUNCTION_GROUP_ARRAY_MAX_ARRAY_SIZE 0xFFFFFF
```
Comment (alexey-milovidov) [inline dbms/src/AggregateFunctions/AggregateFunctionMovingSum.h]: Duplicate name of macro.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
I have questions in https://t.me/joinchat/B44Vi1T-PQA97S02SBwXIw

**Comment by inv2004** [comment]
> I have questions in https://t.me/joinchat/B44Vi1T-PQA97S02SBwXIw

Functions created : groupArrayMovingSum and groupArrayMovingAvg

**Pull request merged**